File: Data Structures & Algorithms/three-integer-sum/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    threeSum(nums) {
        nums.sort((a, b) => a - b);
        const result = [];

        for (let i = 0, n = nums.length; i < n - 2; i++) {
            if (i > 0 && nums[i] === nums[i - 1]) continue;
            
            let l = i + 1;
            let r = n - 1;

            while (l < r) {
                const sum = nums[i] + nums[l] + nums[r]

                if (sum > 0) {
                    r--;
                } else if (sum < 0) {
                    l++;
                } else {
                    result.push([nums[i], nums[l], nums[r]]);
                    r--;
                    l++;
                    while (l < r && nums[l] === nums[l - 1]) {
                        l++;
                    }
                }
            }
        }
        return result;
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l, r = 0, 1
        currMax = nums[0]

        while r < len(nums):
            if nums[r] > currMax:
                currMax = nums[r]
                l += 1
                nums[l] = nums[r]
            
            r += 1
        
        return l + 1